addons/l10n_ar/models/account_journal.py

```diff
@@ -0,0 +1,216 @@
+        expo_codes = ['19', '20', '21']
+        if self.type != 'sale':
+            return []
+        elif self.l10n_ar_afip_pos_system in ['RAW_MAW', 'RLI_RLM', 'II_IM']:
+            return usual_codes + receipt_codes
+        elif self.l10n_ar_afip_pos_system in ['BFERCEL', 'BFEWS']:
+            return usual_codes
+        elif self.l10n_ar_afip_pos_system in ['FEERCEL', 'FEEWS', 'FEERCELP']:
+            return expo_codes
+
+    def get_document_type_sequence(self, invoice):
+        """ Return the match sequences for the given journal and invoice
+        """
+        self.ensure_one()
+        if self.l10n_latam_country_code != 'AR':
+            return False
+        if self.l10n_ar_share_sequences:
+            return self.l10n_ar_sequence_ids.filtered(
+                lambda x: x.l10n_ar_letter == invoice.l10n_ar_letter)
+
+        return self.l10n_ar_sequence_ids.filtered(
+            lambda x: x.l10n_latam_document_type_id ==
+                invoice.l10n_latam_document_type_id)
+
+    def create_document_type_sequences(self):
+        """ Create new sequences for document types, update if can be updated
+        """
+        self.ensure_one()
+        if self.company_id.country_id.code != 'AR':
+            return super().create_document_type_sequences()
+        if not self.type == 'sale':
+            return False
+        if not self.l10n_latam_use_documents:
+            return False
+
+        sequences = self.l10n_ar_sequence_ids
+        sequences.unlink()
+
+        # Create Sequences
+        letters = self.get_journal_letter()
+        internal_types = ['invoice', 'debit_note', 'credit_note']
+        domain = [
+            ('country_id.code', '=', 'AR'),
+            ('internal_type', 'in', internal_types),
+            '|', ('l10n_ar_letter', '=', False),
+            ('l10n_ar_letter', 'in', letters),
+        ]
+        codes = self.get_journal_codes()
+        if codes:
+            domain.append(('code', 'in', codes))
+        documents = self.env['l10n_latam.document.type'].search(domain)
+        sequence_obj = self.env['ir.sequence']
+        for document in documents:
+            if self.l10n_ar_share_sequences:
+                letter_sequence = sequence_obj.search([
+                    ('l10n_ar_letter', '=', document.l10n_ar_letter)])
+                if letter_sequence:
+                    continue
+
+            sequences |= self.env['ir.sequence'].create(
+                document.get_document_sequence_vals(self))
+        return sequences
+
+    @api.constrains('type', 'l10n_ar_afip_pos_system',
+                    'l10n_ar_afip_pos_number', 'l10n_ar_share_sequences')
+    def check_afip_configurations(self):
+        """ IF AFIP Configuration change try to review if this can be done
+        and then create / update the document sequences """
+        for rec in self:
+            invoices = self.env['account.invoice'].search([
+                ('journal_id', '=', rec.id),
```

Comment:
```suggestion
                ('journal_id', 'in', self.ids),
```
To avoid the next raise in the `for`